classes/ReassuranceActivity.php

```diff
@@ -187,7 +187,7 @@ public static function getAllBlockByStatus($id_lang = 1, $id_shop = 1)
 
         foreach ($result as &$item) {
             $item['is_svg'] = !empty($item['custom_icon'])
-                && (self::getMimeType(str_replace(__PS_BASE_URI__, _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR, $item['custom_icon'])) == 'image/svg');
+                && (self::getMimeType(preg_replace('/\\'.__PS_BASE_URI__.'/', _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR, $item['custom_icon'], 1)) == 'image/svg');
```

Comment:
Sorry for late answer. Seems good. Maybe we can make the code even more clear to help reading.
```
        foreach ($result as &$item) {
            $itemIconPath = preg_replace(
                '/\\'.__PS_BASE_URI__.'/',
                _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR,
                $item['custom_icon'],
                1
            );
            $item['is_svg'] = !empty($item['custom_icon'])
                && (self::getMimeType($itemIconPath) == 'image/svg');
        }
```

Also... basically we are aiming to transform `$item['custom_icon']` into a filepath, right? Cannot we do it better than relying on str_replace or preg_replace? It's seems overkill to me to do this 😄

Reply:
We could also just test for `if (__PS_BASE_URI__ == '/') and handle this case alone, then rely on str_replace.

Reply:
Yes it is overkill. Worry is that `$item['custom_icon']` value is not always consistent. With a `if`, there is more code maybe. In all case, it's actually a bug so it need to be correct.